### ISSUE TITLE
Can't reproduce videos with SM open

### ISSUE THREAD

**Issue by latot** [stepmania/stepmania#1245]
Hi all, well, i found this problem, basically i can't reproduce videos when SM is open, i try with chrome for online videos and mpv for local videos, some times works but the most of cases the videos are stalled or without audio, the log of mpv show this:

```
[ao/pulse] The stream is suspended. Bailing out.
```

I use Gentoo 64 (Linux)

Thx

**Comment by kyzentun** [comment]
I suspect this is some weird sound device locking problem.  No clue how to solve it, but videos and sounds from other stuff doesn't play while stepmania is using the sound device.

**Comment by latot** [comment]
I think here clues!

https://github.com/stepmania/stepmania/issues/1385
https://github.com/stepmania/stepmania/issues/1386
https://github.com/stepmania/stepmania/issues/1387

Personally seems SM don't is using the right "protocol" (i don't know if i use the right word) to configure the audio (and other things), and maybe for this this happen.

@kyzentun what do you think?

Thx. Cya.

**Comment by kyzentun** [comment]
I think I don't care because I'll never afford a second monitor, and attempts to fix problems when some random user has to test every attempted fix go nowhere when the user loses interest and leaves.

Given that Stepmania's sound code is over 10 years old, driver changes that deprecate everything stepmania does are likely.

Using libsoundio for the sound backend instead of the current stuff might help, and might have its own unique set of problems.

**Comment by latot** [comment]
Hi, sorry i think i don't explain part of my intentions, part of that issues are only for context, because it feels like if sm is really accessing correctly to the audio configs for example should now affect like that issues, specially for this issue, https://github.com/stepmania/stepmania/issues/1386, so for that i think check that issue can help to solve this irregularity.

Thx. Cya.

**Comment by latot** [comment]
ups, s/now/not/

**Comment by kyzentun** [comment]
I agree that stepmania's audio backgend is probably the problem.  The best solution is for someone to spend a month or two reading libsoundio docs and implementing a completely new backend.  It's not going to be me because I'm not doing stepmania code anymore.

**Comment by latot** [comment]
@kyzentun, how can i know what sound library is using sm actually?, (the priorities), because when i configure it shows i have pulseaudio, alsa, jack and oss, and then it says i have at least one so don't worry about that and i check the linked libs and sm compile with all available libs, but in execution time what lib is using?, or some way to know it, i can't find that info in the logs...

**Comment by latot** [comment]
nothing, i found it, maybe this can handled in other ways, at least temporally.

**Comment by latot** [comment]
Hi, now i'm lost again, the data i can get is ```Sound Driver: Alsa-sw```, but now i'm not sure if that is Alsa or other, because i disable all except pulse but i still get that, so, sorry ask this again, but how i can know what lib is using sm in real time?

**Comment by kyzentun** [comment]
src/arch/arch_default.h controls what order sound drivers are tried in.
```
arch/arch_default.h:61:#define DEFAULT_SOUND_DRIVER_LIST "ALSA-sw,OSS,JACK,Pulse,Null"
```

The first one in that list that succeeds is used.  The sound driver cannot be changed while stepmania is running, it can only be set by changing Preferences.ini before starting Stepmania.

Stepmania logs the sound driver it's using at startup.  Look for a line like this:
```
00:02.556: Sound driver: ALSA-sw
```

**Comment by latot** [comment]
Nice, that helps a lot!, and the idea of libsoundio is good, the only think i'm little worried is only supports Windows7+ ....

Reading the doc there is comparison between this, and other option is for example sdl2, have less options but don't have the windows limit.

What do you think?

**Comment by shakesoda** [comment]
Doesn't matter. We don't support Windows <7 for any future builds now that
5.0 is done.


- Colby

**Comment by latot** [comment]
What do you think to change the priorities to this:
```
"JACK,Pulse,ALSA-sw,OSS,Null"
```

I think make sense with the actual systems, and the code check if are available the options, so for ex, if i don't have started jack (but i compiled with it) sm uses Pulse.

That change can fix two issues.

**Comment by shakesoda** [comment]
ALSA-sw is preferred because Pulse has historically had lag problems.


- Colby

**Comment by latot** [comment]
mm, in my experience i never had that problems....

or maybe when that problems happen?

or, even if historically, actually is relative fixed? (or in your experience, when was the last time you saw this?)

if is no possible this, what do you think can be a good fix for this?, (lock all audio only to keep open sm seems to me a bad idea ...)

Thx. Cya.

**Comment by latot** [comment]
and other point, this is the same or at least can be fixed with this https://github.com/stepmania/stepmania/issues/1386

in some way with that i think we can't implement other things....... (like sdl2 or libsoundio) because that calls pulse for example and we will be in that problem..........................................!!

D:

**Comment by latot** [comment]
Hi, @shakesoda, what do you think about the last 2 comments?

And a middle question (in case i confuse something), about pulse you mean the written code of pulse, or pulse it self?

Thx. Cya.
